app/src/main/java/org/mozilla/fenix/library/downloads/viewholders/DownloadsListItemViewHolder.kt

```diff
@@ -24,24 +28,78 @@ class DownloadsListItemViewHolder(
 
     private var item: DownloadItem? = null
 
+    init {
+        setupMenu()
+
+        itemView.delete_downloads_button.setOnClickListener {
+            val selected = selectionHolder.selectedItems
+            if (selected.isEmpty()) {
+                downloadInteractor.onDeleteAll()
+            } else {
+                downloadInteractor.onDeleteSome(selected)
+            }
+        }
+    }
+
     fun bind(
-        item: DownloadItem
+        item: DownloadItem,
+        showDeleteButton: Boolean,
+        mode: DownloadFragmentState.Mode,
+        isPendingDeletion: Boolean = false
     ) {
-        itemView.download_layout.visibility = View.VISIBLE
+        if (isPendingDeletion) {
+            itemView.download_layout.visibility = View.GONE
+        } else {
+            itemView.download_layout.visibility = View.VISIBLE
+        }
         itemView.download_layout.titleView.text = item.fileName
         itemView.download_layout.urlView.text = item.size.toLong().toMegabyteOrKilobyteString()
 
+        toggleTopContent(showDeleteButton, mode === DownloadFragmentState.Mode.Normal)
```

Comment:
= = = ? what does the triple = do :O

Reply:
Not sure if this is necessary here (re: our convo earlier). Let's just leave this as-is for now.

Reply:
Here you can find some docs around the equality [operators](https://kotlinlang.org/docs/reference/equality.html#referential-equality), in summary `===` is comparing based on reference, vs `==` that compare based on [equals](https://kotlinlang.org/docs/reference/equality.html#equality), in this case either will work as`DownloadFragmentState.Mode.Normal` [is an object](https://github.com/mozilla-mobile/fenix/blob/0ee7e1c28de2a51379c672a0fc084e6bd2ff7105/app/src/main/java/org/mozilla/fenix/library/downloads/DownloadFragmentStore.kt#L54) it has only one reference (it's a singleton), I would prefer if we could use `==` as it will have the same effect and avoid the inherited complexity of `===`.